RunMe.ipynb

```diff
@@ -0,0 +1,282 @@
+    "CACM_tokens = np.array([188887, 85151])\n",
+    "CACM_vocab = np.array([9238, 6334])\n",
+    "heap = HeapRegression(CACM_tokens, CACM_vocab)\n",
+    "parameters = heap.calculate_regression()\n",
+    "print(\"The Heap law parameters are:\")\n",
+    "print(\"(b, k) = {}\".format(parameters))\n",
+    "print(\"For 1 million tokens there would be (by Heap law) {} vocabulary\"\n",
+    "    .format(heap.calculate_vocab(1000000, parameters)))\n",
+    "\n",
+    "graph = FrequencyRankGraph(cacmindex.get_freq())\n",
+    "graph.draw_graph()\n",
+    "graph.draw_log_graph()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### CS276\n",
+    "Calculate tokens and vocabulary size for CS276 collection, and draw its frequency-rank graph.\n",
+    "It will take around 10 minutes."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from CS276Index import *\n",
+    "from heapRegression import *\n",
+    "\n",
+    "cs276index = CS276Index()\n",
+    "cs276index.build()\n",
+    "# cs276index.get_size()\n",
+    "\n",
+    "print()\n",
+    "#following data is the result of size number for full and half text.\n",
+    "CS276_tokens = np.array([25527977,  12796571])\n",
+    "CS276_vocab = np.array([284418, 140665])\n",
+    "heap = HeapRegression(CS276_tokens, CS276_vocab)\n",
+    "parameters = heap.calculate_regression()\n",
+    "print(\"The Heap law parameters are:\")\n",
+    "print(\"(b, k) = {}\".format(parameters))\n",
+    "print(\"For 1 million tokens there would be (by Heap law) {} vocabulary\"\n",
+    "    .format(heap.calculate_vocab(1000000, parameters)))\n",
+    "\n",
+    "graph = FrequencyRankGraph(cs276index.get_freq())\n",
+    "graph.draw_graph()\n",
+    "graph.draw_log_graph()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Part 2 : Inverted index\n",
+    "We create directly the inverted index with frequency. Each entry is of form : \n",
+    "\n",
+    "$\\{term_{id} : [[doc_{id}, frequency], [doc_{id}, frequency], ...], ...\\}$\n",
+    "### CACM\n",
+    "Let's create CACM inverted index"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from BSBIndex import *\n",
+    "\n",
+    "CACMIndex = CACMIndex()\n",
+    "CACMIndex.build()\n",
+    "index = BSBIndex('CACM', CACMIndex.get_term_dict(), CACMIndex.get_document_dict())\n",
+    "index.build()\n",
+    "# print(index.get_index())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Part 3 : Search\n",
+    "### Boolean search"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "from boolean.booleanEvaluation import *\n",
```

Comment:
J'ai une erreur car mon index CACM n'a pas été sauvegardé dans la partie précédente

Reply:
Comment on fait pour sauvegarder ? je trouve pas la ligne correspondante

Reply:
En fait je suis aps sure de comprendre la différencre entre le main de BSBI et le main de BSBIndex. Lequel fait quoi ?

Reply:
C'est justement BSBI le fichier de base qui va sauvegarder les indexes, et BSBI fait appel à BSBIndex. En fait BSBIndex construir concrètement l'index et BSBI est la brique qui englobe le tout car pour CS276 on procède bloc par bloc. Donc BSBI va faire une boucle for pour appliquer BSBIndex sur chaque bloc, puis à la fin tout sauvegarder.